elf/elf.go

```diff
@@ -257,7 +322,21 @@ func elfReadMaps(file *elf.File) (map[string]*Map, error) {
 			mapCount := len(data) / C.sizeof_struct_bpf_map_def
```

Comment:
Actually, since we use the map name included in the name of the ELF section (e.g. `SEC("maps/tcp_event_ipv4")`), we cannot have several `struct bpf_map_def` in one ELF section because we would not know how to distinguish their names.

The idea to put several map definitions in the same ELF section comes from iproute2:
https://git.kernel.org/pub/scm/linux/kernel/git/shemminger/iproute2.git/tree/lib/bpf.c?h=v4.11.0#n1637
They have a single section "maps" for all the maps:
```
		if (data.sec_hdr.sh_type == SHT_PROGBITS &&
		    !strcmp(data.sec_name, ELF_SECTION_MAPS))
			ret = bpf_fetch_maps_begin(ctx, i, &data);
```
They take the map name from the symbol name in the symbol table instead of using the section name like gobpf.

gobpf should either do the same, or stop pretending to support several map definitions in the same ELF section (and check that `len(data) == C.sizeof_struct_bpf_map_def` in the latter case).